custom/icds_reports/ucr/expressions.py

```diff
@@ -288,10 +293,36 @@ class GetAppVersion(JsonObject):
     def configure(self, app_version_string):
         self._app_version_string = app_version_string
 
-    def __call__(self, item, context=None):
+    def get_version_from_app_object(self, item):
+        domain_name = item.get('domain')
+        form_accessor = FormAccessors(domain_name)
+        form_id = item.get('form').get('meta').get('instanceID')
+        form = form_accessor.get_form(form_id)
+
+        app = get_app(domain_name, form.app_id)
```

Comment:
`get_build_by_version` would likely be faster, since it doesn't return the entire app doc but does include `upstream_version`. Not sure if performance matters since this is india only.

Reply:
hey `get_build_by_version` requires the `version` to pass as argument. which we won't have in advance

Reply:
Can't you pull it from the string as you're doing for regular apps?

Reply:
if this would be faster it is a good change to make, but won't block on it

Reply:
Oh yep. Great idea. I can do that. and I see its way faster.